WordPress/openverse-frontend
# Use pnpm (and fix Storybook issues as a result)

## Fixes


Fixes #483 by @krysal 

## Description


Replaces `npm` with `pnpm`. `pnpm` has some significant advantages compared to `npm`, including super fast installs thanks to its local dependency cache. Dependencies are sym-linked into `node_modules` which makes using `pnpm` for many projects ultimately result in smaller disk usage. It also makes dependency re-installation super fast as it just has to create the sym-links. Similarly, deleting `node_modules` is much faster as you're just removing sym-links.

To support Vue we need to use the `shamefully-hoist` option as noted in this issue: https://github.com/pnpm/pnpm/issues/831 (note that the `shamefully-flatten` option has been renamed to `shamefully-hoist`)

Tagging @rbadillap specifically for review on this one as it touches the newly minted Dockerfile 🙂 

Marking this as high priority as not being able to properly use Storybook is not good for redesign development which we're currently moving quickly on.

## Testing Instructions

Checkout this branch and remove `node_modules` (for example `rm -rf node_modules`). Follow the instructions in the readme to install `pnpm` (`volta install pnpm`) then run `pnpm install`. Next run `pnpm storybook` and `pnpm dev` to ensure that they work 🎉 Also run `pnpm build` followed by `pnpm start`.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
